core-bundle/src/Resources/contao/elements/ContentMedia.php

```diff
@@ -125,27 +125,31 @@ protected function compile()
 			$arrFiles = array('m4a'=>null, 'mp3'=>null, 'wma'=>null, 'mpeg'=>null, 'wav'=>null, 'ogg'=>null);
 		}
 
-		$objFiles->reset();
-
 		// Convert the language to a locale (see #5678)
 		$strLanguage = LocaleUtil::formatAsLocale($objPage->language);
 
+		// Determine caption
+		$strCaption = $this->playerCaption;
+
 		// Pass File objects to the template
-		while ($objFiles->next())
+		foreach ($objFiles as $objFile)
```

Comment:
Using `$objFile` for both the model and the file object on line 151 can lead to confusion I think.
How about naming it `$objFileModel`?
```suggestion
		foreach ($objFiles as $objFileModel)
```

Reply:
Done in db6d12026